lib/index.js

```diff
@@ -14,33 +19,34 @@ const GRAMMAR_SCOPES = [
   'text.html.ruby'
 ];
 
-export const config = {
-  executablePath: {
-    title: 'Executable Path',
-    description: 'HTMLHint Node Script Path',
-    type: 'string',
-    default: path.join(__dirname, '..', 'node_modules', 'htmlhint', 'bin', 'htmlhint')
-  }
-};
-
-let executablePath = '';
+export const config = {};
 
 export function activate() {
   require('atom-package-deps').install('linter-htmlhint');
+}
 
-  executablePath = atom.config.get('linter-htmlhint.executablePath');
-
-  atom.config.observe('linter-htmlhint.executablePath', newValue => {
-    executablePath = newValue;
-  });
+function getConfig(filePath) {
+  return findAsync(path.dirname(filePath), '.htmlhintrc')
+    .then(configPath => {
```

Comment:
I'm not sure I understand what you mean. Are you concerned about replicating existing htmlhint functionality inside this package? Since htmlhint does not provide an exported `getConfig` function we're going to need to replicate that functionality in some way.

cosmicconfig could fill this role, but I have two concerns about it. First, it's another module to load which will likely add a few milliseconds to the startup time. Second, it looks for other possible versions of config files as well (`.htmlhintrc.yml`, `.htmlhintrc.js`, etc). htmlhint only supports `.htmlhintrc`. Supporting other forms of the config file might be helpful to someone but strictly speaking it's not how htmlhint behaves and might be confusing to users.

As for the current method, file loading and JSON parsing seem pretty tame as far as adding functionality goes. I'd say we should move it out into another module (or at least file) if there was much more of it, but it's only 18 lines and there's an overhead to using `require`.

If there's some other package that can replicate the behavior of the [original `getConfig`](https://github.com/yaniswang/HTMLHint/blob/master/bin/htmlhint#L297) I'm all for it, but I don't know of one.

Thoughts?